.github/workflows/cpp.yml

```diff
@@ -0,0 +1,59 @@
+name: C++ Tests
+
+on:
+  push:
+    branches: [ main, 'pr/*' ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  clang-format:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - uses: DoozyX/clang-format-lint-action@v0.14
+      with:
+        source: '.'
+        extensions: 'h,hpp,cpp'
+        clangFormatVersion: 9
+    - uses: DoozyX/clang-format-lint-action@v0.14
+      with:
+        source: '.'
+        extensions: 'h,hpp,cpp'
+        clangFormatVersion: 14
+
+  test-cpu:
+    runs-on: ubuntu-latest
+    container: ghcr.io/wdmapp/oneapi-dpcpp-ubuntu-22.04:20230104
+    env:
+      DEBIAN_FRONTEND: noninteractive
+      LD_LIBRARY_PATH: /opt/intel/oneapi/tbb/2021.8.0/env/../lib/intel64/gcc4.8:/opt/intel/oneapi/mpi/2021.8.0//libfabric/lib:/opt/intel/oneapi/mpi/2021.8.0//lib/release:/opt/intel/oneapi/mpi/2021.8.0//lib:/opt/intel/oneapi/mkl/2023.0.0/lib/intel64:/opt/intel/oneapi/debugger/2023.0.0/gdb/intel64/lib:/opt/intel/oneapi/debugger/2023.0.0/libipt/intel64/lib:/opt/intel/oneapi/debugger/2023.0.0/dep/lib:/opt/intel/oneapi/compiler/2023.0.0/linux/lib:/opt/intel/oneapi/compiler/2023.0.0/linux/lib/x64:/opt/intel/oneapi/compiler/2023.0.0/linux/lib/oclfpga/host/linux64/lib:/opt/intel/oneapi/compiler/2023.0.0/linux/compiler/lib/intel64_lin
+      PATH: /opt/intel/oneapi/mpi/2021.8.0//libfabric/bin:/opt/intel/oneapi/mpi/2021.8.0//bin:/opt/intel/oneapi/mkl/2023.0.0/bin/intel64:/opt/intel/oneapi/dev-utilities/2021.8.0/bin:/opt/intel/oneapi/debugger/2023.0.0/gdb/intel64/bin:/opt/intel/oneapi/compiler/2023.0.0/linux/lib/oclfpga/bin:/opt/intel/oneapi/compiler/2023.0.0/linux/bin/intel64:/opt/intel/oneapi/compiler/2023.0.0/linux/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+      CMAKE_BUILD_PARALLEL_LEVEL: 8
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: true
+    - name: clinfo
+      run: |
+        mkdir -p /etc/OpenCL/vendors
+        echo "libintelocl.so" > /etc/OpenCL/vendors/intel-cpu.icd
+        clinfo
+    - name: sycl-ls
+      run: |
+        which sycl-ls
+        sycl-ls
+    - name: setup compiler env
+      run: |
+        which icpx
+        echo "CXX=$(which icpx)" >> $GITHUB_ENV
+    - name: env check
+      run: |
+        env | grep oneapi
+    - name: cmake
+      run: cmake -S . -B build-sycl -DCMAKE_CXX_FLAGS="-fsycl -O2 -g"
```

Comment:
can we tests only with `-fp-model precise` ? less confident for other flags

Reply:
good point, changed.